### PR TITLE
Setup WCK AMI

### PR DESCRIPTION
**Pull request by sienkin**
Initial commit for a WCK AMI
Builds a single AMI suitable for both frontend and backend use, through the use of deployment scripts called when resulting instances are provisioned.

### CODE REVIEW

File: ansible/host_vars/wck.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  - key: relayhost
+    value: smtp-outbound.sharedservices.aws.internal
+
+postfix_net_config:
+  - key: inet_interfaces 
+    value: all
+  - key: inet_protocols 
+    value: ipv4
+
+## Nagios Variables
```
Comment (MartyFox) [inline ansible/host_vars/wck.yaml:37]: Can probably strip out the Nagios code, this is being replaced with CloudWatch going forward.

File: ansible/roles/wck-app-config/files/deployment-scripts/backend_deployment.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
+---
+  - name: Deploy EWF Backend artifact
```
Comment (MartyFox) [inline ansible/roles/wck-app-config/files/deployment-scripts/backend_deployment.yml:2]: EWF -> WCK

File: ansible/roles/wck-app-config/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  vars:
+    ansible_python_interpreter: /usr/bin/python2.6
+  copy:
+    src: "{{ item }}"
+    dest: "/{{ item }}"
+    mode: 0755
+  loop:
+    - etc/httpd/conf/httpd.conf.j2
+    - etc/httpd/conf/ewf_startup.pl
+    - etc/httpd/conf.d/ewf_perl.conf.j2
```
Comment (MartyFox) [inline ansible/roles/wck-app-config/tasks/main.yml]: Is the frontend here based on EWF?

File: ansible/test/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# CHD AMI testing
```
Comment (MartyFox) [inline ansible/test/README.md]: CHD -> WCK

File: packer/build.pkr.hcl [hunk 5]
```diff
@@ -7,7 +7,9 @@ build {
     host_alias = "${var.ansible_host_alias}"
     playbook_file = "${var.playbook_file_path}"
     extra_arguments  = [
-      "-e", "aws_region=${var.aws_region}"
+      "-e", "aws_region=${var.aws_region}",
+      "-e", "s3_bucket=${var.aws_s3_release_bucket}",
+      "-e", "nagios_api_key=${var.nagios_api_key}"
```
Comment (MartyFox) [inline packer/build.pkr.hcl:12]: Can remove Nagios key variable.

File: ansible/playbook.yml [hunk 6]
```diff
@@ -1 +1,16 @@
----
+---  
+- hosts: wck
```
Comment (MartyFox) [inline ansible/playbook.yml:2]: Can remove this role now.

File: packer/variables.pkr.hcl [hunk 7]
```diff
@@ -74,11 +75,29 @@ variable "ssh_private_key_file" {
...
+  type        = string
+  default     = null
+  description = "KMS key ID, arn or alias to use for root volume encryption in the main region. If encrypt_boot is true and this is left null, the AWS default key is used"
+}
+
+variable "nagios_api_key" {
+  type        = string
+  default     = ""
+  description = "This key will be supplied to the Nagios agent Ansible role to populate jinja templates"
+}
```
Comment (MartyFox) [inline packer/variables.pkr.hcl:103]: Nagios var can be removed now

File: ansible/roles/wck-app-config/files/deployment-scripts/backend_deployment.yml (hunk 2 above)
Comment (sienkin) [reply ansible/roles/wck-app-config/files/deployment-scripts/backend_deployment.yml:2]: The backend deployment uses the EWF backend archive. Only the frontend will use a WCK-specific archive.

File: ansible/host_vars/wck.yaml (hunk 1 above)
Comment (sienkin) [reply ansible/host_vars/wck.yaml:37]: As discussed, Nagios config will be retained for now and will be reviewed in the future as necessary.

File: ansible/roles/wck-app-config/tasks/main.yml (hunk 3 above)
Comment (sienkin) [reply ansible/roles/wck-app-config/tasks/main.yml]: Fixed filenames in the task

File: ansible/test/README.md (hunk 4 above)
Comment (sienkin) [reply ansible/test/README.md]: Updated in latest commit

File: packer/build.pkr.hcl (hunk 5 above)
Comment (sienkin) [reply packer/build.pkr.hcl:12]: As discussed, Nagios config will be retained for now and will be reviewed in the future as necessary.

File: ansible/playbook.yml (hunk 6 above)
Comment (sienkin) [reply ansible/playbook.yml:2]: As discussed, Nagios config will be retained for now and will be reviewed in the future as necessary.

File: packer/variables.pkr.hcl (hunk 7 above)
Comment (sienkin) [reply packer/variables.pkr.hcl:103]: As discussed, Nagios config will be retained for now and will be reviewed in the future as necessary.

**Review by MartyFox — APPROVED** [review]

**Pull request merged**